jenkins/spark-tests.sh

```diff
@@ -255,6 +256,59 @@ run_iceberg_tests() {
  fi
 }
 
+run_iceberg_s3tables_test() {
+ # Currently we only support Iceberg 1.6.1 for Spark 3.5.x
+   ICEBERG_VERSION=1.6.1
+   # get the major/minor version of Spark
+   ICEBERG_SPARK_VER=$(echo "$SPARK_VER" | cut -d. -f1,2)
+   IS_SPARK_35X=0
+   # If $SPARK_VER starts with 3.5, then set $IS_SPARK_35X to 1
+   if [[ "$ICEBERG_SPARK_VER" = "3.5" ]]; then
+     IS_SPARK_35X=1
+   fi
+
+  # RAPIDS-iceberg only support Spark 3.5.x yet
+  if [[ "IS_SPARK_35X" -ne "1" ]]; then
+    echo "!!!! Skipping Iceberg tests. GPU acceleration of Iceberg is not supported on $ICEBERG_SPARK_VER"
+  else
+    echo "!!! Running iceberg test with s3table catalog"
+
+    JARS="org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:1.6.1,\
+    software.amazon.s3tables:s3-tables-catalog-for-iceberg-runtime:0.1.6,\
+    software.amazon.awssdk:apache-client:2.29.26,\
+    software.amazon.awssdk:aws-core:2.29.26,\
+    software.amazon.awssdk:dynamodb:2.29.26,\
+    software.amazon.awssdk:glue:2.29.26,\
+    software.amazon.awssdk:http-client-spi:2.29.26,\
+    software.amazon.awssdk:kms:2.29.26,\
+    software.amazon.awssdk:s3:2.29.26,\
+    software.amazon.awssdk:sdk-core:2.29.26,\
+    software.amazon.awssdk:sts:2.29.26,\
+    software.amazon.awssdk:url-connection-client:2.29.26,\
+    software.amazon.awssdk:s3tables:2.29.26,\
+    org.apache.hadoop:hadoop-aws:3.3.4,\
+    com.amazonaws:aws-java-sdk-bundle:1.12.709"
+
+     # aws configure
```

Comment:
this must be either doc in https://github.com/NVIDIA/spark-rapids/blob/branch-25.08/integration_tests/README.md or provide a script prepare&cleanup ENV for developer usages